### PR TITLE
feat: support enable ipv6 

### PR DESCRIPTION
**Pull request by withlin**
fixes: https://github.com/kube-logging/logging-operator/pull/1844 https://github.com/kube-logging/logging-operator/pull/1836

### CODE REVIEW

**Comment by withlin** [comment]
hi @pepov  @csatib02 ,PTAL!!! thanks

**Comment by csatib02** [comment]
Hi @withlin,

Thank you for your contribution, please check out the failing tests.

File: pkg/resources/fluentd/config.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ var fluentdInputTemplate = `
     metrics_path {{ .Monitor.Path }}
 {{- end }}
 </source>
+{{{{ if .Monitor.EnabledIPv6 }}}}
```
Comment (pepov) [inline pkg/resources/fluentd/config.go:66]:
```suggestion
{{- if .Monitor.EnabledIPv6 }}
```

File: pkg/resources/fluentd/configsecret.go [hunk 2]
```diff
@@ -30,9 +30,10 @@ type fluentdConfig struct {
 	LogFormat string
 	LogLevel  string
 	Monitor   struct {
-		Enabled bool
-		Port    int32
-		Path    string
+		Enabled     bool
+		EnabledIPv6 bool
```
Comment (pepov) [inline pkg/resources/fluentd/configsecret.go:34]: have you considered using enabledIPv6 by default if it is configured for fluentbit?

File: pkg/sdk/logging/api/v1beta1/fluentd_types.go [hunk 3]
```diff
@@ -51,6 +51,7 @@ type FluentdSpec struct {
 	TLS                    FluentdTLS        `json:"tls,omitempty"`
 	Image                  ImageSpec         `json:"image,omitempty"`
 	DisablePvc             bool              `json:"disablePvc,omitempty"`
+	EnabledIPv6            bool              `json:"enabledIPv6,omitempty"`
```
Comment (pepov) [inline pkg/sdk/logging/api/v1beta1/fluentd_types.go:54]: I would consider a simple boolean flag here as well

**Review by pepov — COMMENTED** [review]
I would suggest using a boolean flag for both fluentbit/fluentd and use that same flag for the monitor automatically

**Comment by withlin** [comment]
@pepov yeah, agree with on that.  can u reivew again ?

**Comment by withlin** [comment]
@csatib02  can u run pipeline again?

File: pkg/resources/fluentd/config.go [hunk 4]
```diff
@@ -63,6 +63,7 @@ var fluentdInputTemplate = `
     metrics_path {{ .Monitor.Path }}
 {{- end }}
 </source>
+{{- if .Monitor.EnabledIPv6 }}
```
[the remark this answers is not in the document]
Comment (withlin) [reply pkg/resources/fluentd/config.go]: done

**Comment by withlin** [comment]
@pepov  done

**Review by csatib02 — APPROVED** [review]
LGTM!

**Comment by husira** [comment]
@csatib02 are there any plan when this fix will be released?

We have a cluster that only supports ipv4. We currently had to deactivate metrics with release 4.11.x, as otherwise fluentbit and fluentd cannot start, as an ipv6 listener is automatically created for the metrics port.

**Comment by husira** [comment]
Hey @csatib02 

Sounds good. :) Thank you very much for your quick reply - really appreciate it

**Pull request merged**